dataplane/linux/int_dataplane.go

```diff
@@ -1283,7 +1283,7 @@ func (d *InternalDataplane) setUpIptablesBPF() {
 
 		var rawRules []iptables.Rule
 		if t.IPVersion == 4 && rulesConfig.WireguardEnabled && len(rulesConfig.WireguardInterfaceName) > 0 &&
-			rulesConfig.RouteSource == "WorkloadIPs" {
```

Comment:
I think another option here would be to have it like 

`rulesConfig.RouteSource == "WorkloadIPs" && d.config.Wireguard.EncryptHostTraffic`

which should limit the feature to just workloadIPs i.e. not calicoCNI using calicoIPAM. Host-to-host might work with calicoCNI and IPAM perfectly but I guess it just widens the blast radius for testing for 3.9 if we say we have to now test essentially all platforms host-to-host. What do you think?

Reply:
I'd agree with you about the blast radius, how about I add workload IP to [this](https://github.com/projectcalico/felix/pull/2897/files#diff-6d3b325b8dfb02061fd60c6641b100ad40c798ceaa2a27105b29e2af5cfb8847R473) check here, would that be functionally the same?

Reply:
If it defaults to disabled then I think it'd be OK to not require WorkloadIPs, especially if you can kick the tires in CalicoIPAM mode and veryify that it's not totally broken.  You could document that it's only been tested with WorkloadIPs so should be considered "tech preview" in CalicoIPAM mode.

Most users won't switch away from the defaults so if we document it reasonably we might get some community testers who want that feature to work.